via/security.py

```diff
@@ -0,0 +1,39 @@
+from datetime import datetime, timedelta, timezone
+
+import jwt
+from pyramid.security import Allowed, Denied
+
+
+class ViaSecurityPolicy:
+    def permits(self, request, _context, permission):
+        try:
+            encoded_jwt = request.headers["Authorization"][len("Bearer ") :]
+        except KeyError:
+            return Denied("No Authorization header")
+
+        try:
+            jwt.decode(
+                encoded_jwt,
+                self._get_jwt_secret(request),
+                algorithms=["HS256"],
+                options={"require": ["exp"]},
+            )
+        except jwt.InvalidTokenError:
+            return Denied("Invalid JWT")
+
+        if permission == "api":
+            return Allowed("Valid JWT provided")
+
+        return Denied("Unknown permission")
+
+    @classmethod
+    def encode_jwt(cls, request):
+        return jwt.encode(
+            {"exp": datetime.now(tz=timezone.utc) + timedelta(hours=48)},
+            cls._get_jwt_secret(request),
+            algorithm="HS256",
+        )
```

Comment:
We might want to add something more than just the expiry time to the JWT payload. For example I wonder if we could encode the permissions that the JWT grants into the JWT?